tests/utils_tests/test_autoreload.py

```diff
@@ -295,6 +295,41 @@ class MyException(Exception):
             with self.assertRaisesMessage(MyException, 'Test Message'):
                 autoreload.raise_last_exception()
 
+    def test_raises_custom_exception(self):
+        class MyException(Exception):
+            def __init__(self, msg, extra_context):
+                super().__init__(msg)
+                self.extra_context = extra_context
+        # Create the exception
+        try:
+            raise MyException('Test Message', 'extra context')
+        except MyException:
+            exc_info = sys.exc_info()
+
+        with mock.patch('django.utils.autoreload._exception', exc_info):
+            with self.assertRaisesMessage(MyException, 'Test Message'):
+                autoreload.raise_last_exception()
+
+    def test_raises_exception_with_context(self):
+        class MyException(Exception):
+            def __init__(self, number):
+                self.number = number
+
+        try:
+            raise MyException(2)
+        except MyException as e:
+            try:
+                raise MyException(1) from e
+            except MyException:
+                exc_info = sys.exc_info()
+
+        with mock.patch('django.utils.autoreload._exception', exc_info):
+            with self.assertRaises(MyException) as cm:
+                autoreload.raise_last_exception()
+            
+            self.assertEqual(cm.exception.number, 1)
+            self.assertEqual(cm.exception.__cause__.number, 2)
```

Comment:
I think a specialized class is not required for this test case as you can rely on `.args`

```python
try:
    raise Exception(2)
except Exception as e:
    try:
        raise Exception(1) from e
    except Exception:
        exc_info = sys.exc_info()
...

self.assertEqual(cm.exception.args[0], 1)
self.assertEqual(cm.exception.__cause__.args[0], 2)

Reply:
Good idea, fixed!